xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -216,22 +217,24 @@ void CAESinkPipewire::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
   auto loop = pipewire->GetThreadLoop();
   loop->Lock();
 
-  CAEDeviceInfo device;
-  device.m_deviceType = AE_DEVTYPE_PCM;
-  device.m_deviceName = "Default";
-  device.m_displayName = "Default";
-  device.m_displayNameExtra = "Default Output Device (PIPEWIRE)";
-  device.m_wantsIECPassthrough = true;
+  CAEDeviceInfo defaultDevice;
+  defaultDevice.m_deviceType = AE_DEVTYPE_PCM;
+  defaultDevice.m_deviceName = "Default";
+  defaultDevice.m_displayName = "Default";
+  defaultDevice.m_displayNameExtra = "Default Output Device (PIPEWIRE)";
+  defaultDevice.m_wantsIECPassthrough = true;
 
-  for (const auto& format : formatMap)
-    device.m_dataFormats.emplace_back(format.second);
+  std::for_each(formatMap.cbegin(), formatMap.cend(),
+                [&defaultDevice](const auto& pair)
+                { defaultDevice.m_dataFormats.emplace_back(pair.second); });
```

Comment:
i'd keep the range based for loops here because they're much easier to read imo.
in any case `const auto& pair` should be `const auto& format` like it's done for `rate` below.

Reply:
Can't use a range based for loop when there is no `begin` and `end` method available.